### ISSUE TITLE
bazel 0.4.5rc4: error-prone stops building third_party/checker_framework_dataflow

### ISSUE THREAD

**Issue by aehlig** [bazelbuild/bazel#2642]
From the announcement of 0.4.5rc4

```
# Release 0.4.5rc4 (2017-03-06)

Baseline: 2e689c29d5fc8a747216563235e905b1b62d63b0

Cherry picks:
   + a28b54033227d930672ec7f2714de52e5e0a67eb:
     Fix Cpp action caching
   + 6d1d424b4c0da724e20e14235de8012f05c470f8:
     Fix paths of binaries in .deb packages.
   + 0785cbb672357d950e0c045770c4567df9fbdc43:
     Update to guava 21.0 and Error Prone version 2.0.18-20160224
[...]
```
Cherry-picking a library update into a release candidate is a bold strategy. In any case,
the error-prone in 0.4.5rc4 stops `third_party/checker_framework_dataflow` being compiled when bazel builds itself during bootstrap.

```
[...]

ERROR: /memfs/tmp/memfs/tmp/bazel/work/bazel-0.4.5/third_party/checker_framework_dataflow/BUILD:10:1: Java compilation in rule '//third_party/checker_framework_dataflow:checker_framework_dataflow' failed: java failed: error executing command 
  (cd /memfs/tmp/memfs/tmp/bazel/work/bazel_tmp/bazel_bPHPkCOI/out/execroot/bazel-0.4.5 && \
  exec env - \
    LC_CTYPE=en_US.UTF-8 \
  /usr/local/openjdk8/bin/java -Xbootclasspath/p:third_party/java/jdk/langtools/javac-9-dev-r3297-4.jar -XX:+TieredCompilation '-XX:TieredStopAtLevel=1' -jar bazel-out/host/bin/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/bootstrap_deploy.jar @bazel-out/local_freebsd-opt/bin/third_party/checker_framework_dataflow/libchecker_framework_dataflow.jar-2.params): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
third_party/checker_framework_dataflow/java/org/checkerframework/dataflow/cfg/CFGDOTVisualizer.java:264: error: [MissingCasesInEnumSwitch] Non-exhaustive switch; either add a default or handle the remaining cases: CONDITIONAL_BLOCK, SPECIAL_BLOCK
                switch (bb.getType()) {
                ^
    (see http://errorprone.info/bugpattern/MissingCasesInEnumSwitch)
Target //src:bazel failed to build
```

There is an easy fix for this, and I think it should be cherry-picked to the release as well, if we want to have that error-prone update in the release.

```
--- third_party/checker_framework_dataflow/java/org/checkerframework/dataflow/cfg/CFGDOTVisualizer.java.orig    1979-12-31 23:00:00 UTC
+++ third_party/checker_framework_dataflow/java/org/checkerframework/dataflow/cfg/CFGDOTVisualizer.java
@@ -269,6 +269,8 @@ public class CFGDOTVisualizer {
                     case EXCEPTION_BLOCK:
                         lastNode = ((ExceptionBlock) bb).getNode();
                         break;
+                    default:
+                        break;
                 }
                 if (lastNode != null) {
                     sb2.append("\\n~~~~~~~~~\\n");
```

Note: I found that issue on FreeBSD but, as it is a java-compilation, it could be present at all platforms.

**Comment by cushon** [comment]
I suppressed the error for that library, but the fix didn't make it into the RC: 81ae08bbc13f5f4a04f18caae339ca77ae2699c1

**Comment by damienmg** [comment]
As suggested by @aehlig, filed a separate branch for testing that bootstrap. We use to have that test in the Bazel jobs
